art_internal_impl.hpp

```diff
@@ -1060,23 +1336,96 @@ class basic_inode_4 : public basic_inode_4_parent<ArtPolicy> {
   }
   UNODB_DETAIL_RESTORE_MSVC_WARNINGS()
 
-  constexpr void delete_subtree(db &db_instance) noexcept {
+  [[nodiscard]] constexpr node_ptr get_child(
+      std::uint8_t child_index) noexcept {
+    return children[child_index].load();
+  }
+
+  // N4 - position on the first child (there is always at least one child for
+  // N4).
+  [[nodiscard]] constexpr typename basic_inode_4::iter_result begin() noexcept {
+    const auto key = keys.byte_array[0].load();
+    return {node_ptr{this, node_type::I4}, key, static_cast<uint8_t>(0)};
+  }
+
+  // N4 - position on the last child (there is always at least one child for
+  // N4).
+  [[nodiscard]] constexpr typename basic_inode_4::iter_result last() noexcept {
+    const std::uint8_t child_index = this->children_count.load() - 1;
```

Comment:
And everywhere below:
```suggestion
    const auto child_index = this->children_count.load() - 1;
```